vultr.php

```diff
@@ -2249,53 +2122,115 @@ public function tabClientStats($package, $service, array $get = null, array $pos
+                            ];
+                            $this->Services->edit($service->id, $data);
+
+                            if ($this->Services->errors()) {
+                                $vars = (object) $post;
+                                $this->Input->setErrors($this->Services->errors());
+                            }
+                        }
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+
         // Get the server details
         $params = [
             'SUBID' => $service_fields->vultr_subid
         ];
         $this->log('api.vultr.com|list', serialize($params), 'input', true);
 
         if ($package->meta->server_type == 'server') {
-            $server_details = (array) $this->parseResponse($vultr_api->listServers($params));
+            $server_details = $this->parseResponse($vultr_api->listServers($params));
         } else {
-            $server_details = (array) $this->parseResponse($vultr_api->listBaremetal($params));
-        }
-
-        // Get application details
-        $application_details = new stdClass();
-        if (isset($server_details['os']) && trim($server_details['os']) == 'Application') {
-            $application_details = $this->getApplicationDetails($vultr_api, $service);
+            $server_details = $this->parseResponse($vultr_api->listBaremetal($params));
         }
 
         $this->view->set('module_row', $row);
         $this->view->set('package', $package);
         $this->view->set('service', $service);
         $this->view->set('service_fields', $service_fields);
-        $this->view->set('server_details', $server_details);
-        $this->view->set('application_details', $application_details);
+        $this->view->set('templates', $templates);
+        $this->view->set('server_details', (isset($server_details) ? $server_details : new stdClass()));
+        $this->view->set('vars', (isset($vars) ? $vars : new stdClass()));
 
         $this->view->setDefaultView('components' . DS . 'modules' . DS . 'vultr' . DS);
 
         return $this->view->fetch();
     }
 
     /**
-     * Client Snapshots tab.
+     * Snapshots (Admin or Client) tab.
      *
      * @param stdClass $package A stdClass object representing the current package
      * @param stdClass $service A stdClass object representing the current service
-     * @param array $get Any GET parameters
      * @param array $post Any POST parameters
-     * @param array $files Any FILES parameters
+     * @param string $template_name The name of the template to us
      * @return string The string representing the contents of this tab
      */
-    public function tabClientSnapshots($package, $service, array $get = null, array $post = null, array $files = null)
+    private function getTabSnapshots($package, $service, array $post = null, $template_name = 'tab_snapshots')
```

Comment:
It may be better to pass a boolean value of whether to load the client template or not, rather than specifying the exact tab to load, e.g.

 private function getTabSnapshots($package, $service, array $post = null, $client = false)
{
    $template = ($client ? 'tab_client_snapshots' : 'tab_snapshots');
    $this->view = new View($template, 'default');

    ...
}

Reply:
Sure, I suppose there isn't really a chance of their being more than two options.